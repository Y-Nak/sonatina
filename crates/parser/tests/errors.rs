use std::path::Path;

use dir_test::{dir_test, Fixture};
use sonatina_parser::parse_module;
mod common;

#[dir_test(
    dir: "$CARGO_MANIFEST_DIR/test_files/errors/",
    glob: "*.sntn"
)]
fn test_errors(fixture: Fixture<&str>) {
    let Err(errs) = parse_module(fixture.content()) else {
        panic!("expected parse_module to fail with errors");
    };
    let path = Path::new(fixture.path())
        .file_name()
        .unwrap()
        .to_string_lossy();

    let mut v = vec![];
    for err in errs {
        err.print(&mut v, &path, fixture.content(), false).unwrap();
    }
    let s = String::from_utf8(v).unwrap();
    snap_test!(s, fixture.path());
}
