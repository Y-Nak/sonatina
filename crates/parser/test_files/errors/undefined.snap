---
source: crates/parser/tests/errors.rs
expression: s
input_file: crates/parser/test_files/errors/undefined.sntn
---
error: parse error
 --> undefined.sntn:2:25
  |
2 | type %foo = { i8, i16, *%s1 };
  |                         ^^^ undefined type: `%s1`
  |error: parse error
 --> undefined.sntn:6:22
  |
6 |         v0.i8 = call %foo 100.i8;
  |                      ^^^^ undefined function: `%foo`
  |error: parse error
 --> undefined.sntn:7:24
  |
7 |         v2.i8 = add v0 v1;
  |                        ^^ undefined value: `v1`
  |error: parse error
 --> undefined.sntn:8:21
  |
8 |         v3.i8 = add v1 v1;
  |                     ^^ undefined value: `v1`
  |error: parse error
 --> undefined.sntn:8:24
  |
8 |         v3.i8 = add v1 v1;
  |                        ^^ undefined value: `v1`
  |error: parse error
 --> undefined.sntn:9:14
  |
9 |         jump block2;
  |              ^^^^^^ undefined block: `block2`
  |
